src/OpenTelemetry.Extensions.Enrichment/README.md

```diff
@@ -6,6 +6,165 @@
+Activity.SpanId:             bfcbf9d5746009d6
+Activity.TraceFlags:         Recorded
+Activity.ActivitySourceName: MyCompany.MyProduct.MyLibrary
+Activity.DisplayName:        SayHello
+Activity.Kind:               Internal
+Activity.StartTime:          2023-03-20T09:39:21.9642338Z
+Activity.Duration:           00:00:00.0016887
+Activity.Tags:
+    hello: world
+    my key: my value
+Resource associated with Activity:
+    service.name: unknown_service:Examples.Enrichment
+```
+
+### Extension methods
+
+There are a bunch of extension methods available for registering your custom trace
+enricher class. The methods that utilize `IServiceCollection` are tailored for
+Dependency Injection-based situations, allowing for registration of
+`TracerProviderBuilder` and `IServiceCollection` at different locations
+within your codebase. Conversely, the methods relying on `TracerProviderBuilder`
+are meant to be employed in non-DI scenarios.
+
+In case you would like to use a comprehensive enricher class that may require
+injection or interaction with other classes, you may utilize either of these
+two methods. Both methods take a type T parameter to specify the type of your
+enricher class. In this case, the enricher class will be created for you by
+Dependency Injection:
+
+```csharp
+public static AddTraceEnricher<T>(this IServiceCollection services)
+public static AddTraceEnricher<T>(this TracerProviderBuilder builder)
+```
+
+If you prefer to instantiate your enricher class on your own, you may use one of
+these methods which allow for the usage of pre-existing enricher objects:
+
+```csharp
+public static AddTraceEnricher(this IServiceCollection services, TraceEnricher enricher)
+public static AddTraceEnricher(this TracerProviderBuilder builder, TraceEnricher enricher)
+```
+
+If you only need to enrich a small amount of data, it may not be necessary to create
+an enricher class. Instead, you can make use of the following methods which accept
+an `Action<TraceEnrichmentBag>` delegate:
+
+```csharp
+public static AddTraceEnricher(this IServiceCollection services, Action<TraceEnrichmentBag> enrichmentAction)
+public static AddTraceEnricher(this TracerProviderBuilder builder, Action<TraceEnrichmentBag> enrichmentAction)
+```
+
+If you would rather use a factory method to instantiate your enricher class,
+with the possibility of interacting with `IServiceProvider,` you can utilize one
+of these two methods:
+
+```csharp
+public static AddTraceEnricher(this IServiceCollection services, Func<IServiceProvider, TraceEnricher> enricherImplementationFactory)
+public static AddTraceEnricher(this TracerProviderBuilder builder, Func<IServiceProvider, TraceEnricher> enricherImplementationFactory)
+```
+
+## Recommendations and best practices
+
+You can add any number of custom enrichers, but it is advisable to only include
+properties that are truly beneficial to prevent an excessive increase in the
+number of dimensions associated with each `Activity`.
```

Comment:
nit: `dimensions` are more of a metric thing (at least in my mind), maybe just use "tags" or "attributes" here?

Reply:
Updated